### PR TITLE
Rewrite internals of `EffectDrawBase`

### PR DESCRIPTION
**Pull request by sirdoombox**
### Changes
- Internally, `EffectDrawBase` now uses `CompositionCustomVisualHandler`.
  - Whenever animations are disabled, GPU & CPU usage are significantly lower (near 0%) - Related Issue: #167 
- Fix `SukiTransitioningContentControl`.
  - Transition animations now play correctly.
  - Backbuffer is cleared out when it's done playing it's animation and assigning `null` to the content is now valid which should make detaching content from the visual tree work as intended.

### Outstanding Issues
- The `Loading` control - even when invalidating only the rect for it - still has very high GPU utilisation and I'm not quite sure why.
- The `ShaderToyRenderer` in the demo also suffers from this same issue - though it seems to use less GPU for reasons beyond my understanding.
- We don't really have a docs page covering `SukiWindow` at the moment.

I wanted to ideally have at least one or two other people test this to make sure the GPU/CPU issues are in fact under control and everything renders correctly, before this gets merged as it's a pretty major internal change.

### CODE REVIEW

**Comment by kikipoulet** [comment]
No problem on my side, behave like expected with great performance improvements

**Comment by sirdoombox** [comment]
I annoyingly discovered a bug with `SukiTransitioningContentControl` whilst I was testing this out in another app, the first time content was set `IsHitTestVisible` was being set to false and I'm not fully sure why, struggling to actually get it to enable correctly. 

Whilst trying to fix this bug I also discovered that the core functionality was completely disabled and broken in #259 for some reason. I think the bug this was supposed to fix needs a solution, hopefully I can come up with one that retains the functionality of the transitioning content control.

**Comment by sirdoombox** [comment]
I'm not 100% certain I've fixed the issue that #259 was supposed to fix, but it _should_ do the job, now the back buffer is cleared out and you can assign `null` to the `Content `property which should cause the content to be cleared out correctly. It also re-enables the transition animations and such.

**Comment by sirdoombox** [comment]
Could you confirm that #259 isn't regressed with these changes @AuroraZiling? If the crash is back, please send me a repro and I'll try and implement a better solution.

**Review by AuroraZiling — APPROVED** [review]

File: SukiUI/Utilities/Effects/EffectDrawBase.cs
Comment (maxkatz6) [inline SukiUI/Utilities/Effects/EffectDrawBase.cs:99]: Unless you need to invalidate only small portion of the visual, you can just call `Invalidate()` method. It will reduce some overhead on calculating dirty rects per visual.

File: SukiUI/Utilities/Effects/EffectBackgroundDraw.cs
Comment (maxkatz6) [inline SukiUI/Utilities/Effects/EffectBackgroundDraw.cs:31]: You can replace `TransitionSeconds` with `CompositionCustomVisualHandler.CompositionNow` timespan property, which is syncrhonised with render timer. Probably won't improve anything perf-wise though.

**Comment by sirdoombox** [comment]
Thanks for the suggestions @maxkatz6, PR is ready to go.

I won't merge it myself just so one final sanity check gets done before it gets merged.

**Comment by kikipoulet** [comment]
> Thanks for the suggestions @maxkatz6, PR is ready to go.
> 
> I won't merge it myself just so one final sanity check gets done before it gets merged.

I tested it, LGTM

**Pull request merged**
